### PR TITLE
[move-mutator] application skeleton

### LINKED ISSUE #26: Logging subsystem

### PR DESCRIPTION
**Pull request by asmie**
### Description

This PR contains different improvements over basic application skeleton and functionality.

Logs on various levels added to make it easier to debug and trace the application work.

Added comments and minor code improvements.

### Test Plan
Project unit tests.
Tests covered by `README.md` description.

### CODE REVIEW

**Review by Rqnsom — APPROVED** [review]
LGTM (already reviewed [here](https://github.com/eigerco/aptos-core/pull/38#pullrequestreview-1800392160))

File: third_party/move/tools/move-mutator/src/lib.rs [hunk 1]
```diff
@@ -11,41 +15,54 @@ mod report;
...
+/// # Returns
+///
+/// * `anyhow::Result<()>` - Returns `Ok(())` if the mutation process completes successfully, or an error if any error occurs.
 pub fn run_move_mutator(
     options: cli::Options,
     config: BuildConfig,
     package_path: PathBuf,
 ) -> anyhow::Result<()> {
-    println!(
+    pretty_env_logger::init();
```
Comment (vineethk) [inline third_party/move/tools/move-mutator/src/lib.rs:40]: Might be good to document the `RUST_LOG=` env variable in the README.md

File: third_party/move/tools/move-mutator/src/compiler.rs [hunk 2]
```diff
@@ -35,7 +34,7 @@ use move_package::BuildConfig;
 pub fn generate_ast(
     mutator_config: &Configuration,
     config: &BuildConfig,
-    package_path: PathBuf,
+    _package_path: PathBuf,
```
Comment (vineethk) [inline third_party/move/tools/move-mutator/src/compiler.rs:37]: Are we expecting to use this param in the future?

Comment (vineethk) [reply third_party/move/tools/move-mutator/src/compiler.rs:37]: The documented arguments are incorrect, please update.

File: third_party/move/tools/move-mutator/src/compiler.rs [hunk 3]
```diff
@@ -71,12 +72,14 @@ pub fn generate_ast(
         flags,
         &config.compiler_config.known_attributes,
     )
-    .set_interface_files_dir(interface_files_dir)
+    .set_interface_files_dir(interface_files_dir.to_str().unwrap_or("").to_string())
```
Comment (vineethk) [inline third_party/move/tools/move-mutator/src/compiler.rs:75]: Use `expect` instead of `unwrap_or`.

**Review by vineethk — APPROVED** [review]
Minor comments, LGTM.

File: third_party/move/tools/move-mutator/src/lib.rs (hunk 1 above)
Comment (asmie) [reply third_party/move/tools/move-mutator/src/lib.rs:40]: Documented in this [PR](https://github.com/eigerco/aptos-core/pull/60).

File: third_party/move/tools/move-mutator/src/compiler.rs (hunk 3 above)
Comment (asmie) [reply third_party/move/tools/move-mutator/src/compiler.rs:75]: Corrected in this [PR](https://github.com/eigerco/aptos-core/pull/60).

File: third_party/move/tools/move-mutator/src/compiler.rs (hunk 2 above)
Comment (asmie) [reply third_party/move/tools/move-mutator/src/compiler.rs:37]: Updated.

**Pull request merged**